File: 0067-add-binary/0067-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
       int A= a.length()-1;
       int B= b.length()-1;
       int carry=0;
       int sum=0; 
       string ans="";
       while(A>=0 || B>=0 || carry >0)
       {
        sum=carry;
        if(A>=0)
        {
        sum=sum + a[A]-'0';
        A--;
        }

        if(B>=0)
        {sum=sum + b[B]-'0';
        B--;
        }

        ans= char(sum%2 + '0') +ans;
        carry=sum/2;
       }
       return ans;   
    }

};

File: 0067-add-binary/0067-add-binary.java
class Solution {
    public String addBinary(String a, String b) {
        int sum = 0;
        int carr =0;
        String ans = "";
        int i = a.length()-1 ;
        int j = b. length()-1 ; 
        while( i>=0 || j>=0 || carr >0 ){
            sum = carr;
           if ( i >= 0 ){
           sum = sum + a.charAt(i)-'0';
           i--;
           }
           if (j >= 0 ) {
            sum = sum + b.charAt(j)-'0';
            j--;
           }

            ans = sum%2 + ans ;
            carr = sum/2;
        }
    return ans;
    }
}